backend/users/views/user.py

```diff
@@ -232,3 +243,57 @@ def generate_reset_password_token(self, request, *args, **kwargs):
         data = {"reset_password_token": jwt_token}
 
         return ApiResponse(data=data, status=status.HTTP_200_OK, headers=self.get_success_headers({}))
+
+
+class UserAwaitingApprovalViewSet(
+    GenericViewSet,
+    mixins.ListModelMixin,
+):
+    user_model = get_user_model()
+    permission_classes = [IsAdmin]
+    pagination_class = DefaultPageNumberPagination
+    serializer_class = UserAwaitingApprovalSerializer
+    ordering_fields = ["username"]
+    ordering = ["username"]
+    filter_backends = [OrderingFilter, MatchFilter, DjangoFilterBackend]
+    lookup_field = "username"
+    search_fields = ["username"]
+    filterset_class = UserFilter
+
+    def get_queryset(self):
+        return self.user_model.objects.filter(channel=None).exclude(username="deleted")
+
+    def delete(self, request, *args, **kwargs):
+        try:
+            user = User.objects.get(username=request.GET.get("username"))
+            user.delete()
+            return ApiResponse(data={"message": "User removed"}, status=status.HTTP_200_OK)
+        except User.DoesNotExist or User.MultipleObjectsReturned:
+            pass
+        return ApiResponse(data={"message": "User not found"}, status=status.HTTP_404_NOT_FOUND)
+
+    def put(self, request, *args, **kwargs):
+        d = json.loads(request.body)
+        try:
+            user = User.objects.get(username=request.GET.get("username"))
+        except User.DoesNotExist or User.MultipleObjectsReturned:
+            return ApiResponse(data={"message": "User not found"}, status=status.HTTP_404_NOT_FOUND)
+
+        channel_name = get_channel_name(request)
+        channel_data = {"channel_name": channel_name}
+        channel_data["role"] = _validate_role(d.get("role"))
+        channel_data["user"] = user
+        UserChannel.objects.create(**channel_data)
+        user.refresh_from_db()
+        data = UserSerializer(instance=user).data
+        return ApiResponse(data=data, status=status.HTTP_200_OK)  # get success header ?
+
+    # TODO THIS SHOULD NOT BE IN THE PULL REQUEST
+    def post(self, request, *args, **kwargs):
+        d = json.loads(request.body)
+        user, created = User.objects.get_or_create(username=d.get("username"), email=d.get("email"))
+        if created:
+            user.set_password(d.get("password"))
+            user.save()
+        user_json = serialize("json", [user])
+        return ApiResponse({"users_awaiting_approval": user_json})
```

Comment:
Why do we return a list if it's always one element?

Reply:
because for some reason serialize wants an iterable
but this doesn't matter this bits of code is for frontend testing, I'll delete it before merge (I kept a comment of Olivier open for that)